test/e2e/tls_util_test.go

```diff
@@ -197,6 +200,31 @@ func TestNoneOfCaAndAppSecretExist(t *testing.T) {
 	verifyCASecret(t, caSecret, namespace)
 }
 
+// TestCustomCA ensures that if a user provides a custom Key and Cert and the CA and Application TLS assets
+// do not exist, the GenerateCert method can use the custom CA to generate the TLS assest.
+func TestCustomCA(t *testing.T) {
+	f := framework.Global
+	ctx := f.NewTestCtx(t)
+	defer ctx.Cleanup(t)
+	namespace, err := ctx.GetNamespace()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	cg := tlsutil.NewSDKCertGenerator(f.KubeClient)
+
+	ccfg.CAKey = "./testdata/ca.key"
+	ccfg.CACert = "./testdata/ca.crt"
```

Comment:
let's keep the test variable the same for all tests. I think modifying the `ccfg`will cause other tests to error out.

Let's just create a local `*tlsutil.CertConfig` for this test only.